discovery-infra/start_discovery.py

```diff
@@ -166,6 +213,45 @@ def _cluster_create_params():
     return params
 
 
+def _find_free_cidr_machine(start_cidr_machine):
+    cidr_machine = IPNetwork(start_cidr_machine)
+    while _is_cidr_machine_exist(str(cidr_machine)):
+        cidr_machine += 1
+    return str(cidr_machine)
+
+
+def _is_cidr_machine_exist(cidr_machine):
+    cmd_output = utils.run_command(f'/usr/sbin/ip route show {cidr_machine}')
+    is_exist = bool(cmd_output)
+    return is_exist
+
+
+def _find_free_network_bridge():
+    i = 0
+    bridge = f'tt{i}'
+    while _is_bridge_exist(bridge):
+        i += 1
+        bridge = f'tt{i}'
+    return bridge
+
+
+def _is_bridge_exist(bridge):
+    def raise_error_if_occurred(cmd, out, err):
```

Comment:
1. Using the same name is confusing.
2. I don't see a reason for this `callback` to exist. Why not normal try-except with raises?
```python
    try:    
        cmd_output = utils.run_command(
            f'/usr/sbin/ip link show {bridge}',
            shell=False
        )
    except RuntimeError as ex:
        if 'does not exist' in ex.message:
            is_exist = False
        else:
            raise

    is_exist = True
    ....
```

3. After giving it a thought - Why do you even check the output and not the returned value? It would be much simpler and still accurate